docs/writing-benchmarks.md

```diff
@@ -0,0 +1,77 @@
+## Writing Benchmarks
+
+To get started, let's look at a simple example:
+
+```kotlin
+@BenchmarkMode(Mode.Throughput)
+@OutputTimeUnit(TimeUnit.MILLISECONDS)
+@Warmup(iterations = 5, time = 1, timeUnit = TimeUnit.SECONDS)
+@Measurement(iterations = 1, time = 1, timeUnit = TimeUnit.SECONDS)
+@Fork(1)
+@BenchmarkTimeUnit(TimeUnit.MILLISECONDS)
+@State(Scope.Benchmark)
+class ExampleBenchmark {
+
+    @Param("1", "2")
+    var param: Int = 0
+
+    private var list: List<Int> = ArrayList()
+
+    @Setup
+    fun prepare() {
+        list = List(param) { it }
+    }
+
+    @Benchmark
+    fun benchmarkMethod(): Int {
+        return list.sum()
+    }
+
+    @TearDown
+    fun cleanup() {
+        list = ArrayList()
```

Comment:
WDYT about the following changes:
* Change param values to "4" and "10".
* Make `list` a mutable list.
* Fill the `list` in `Setup` function
* Clear the `list` in `Teardown` function

I think the sample will look a bit more like real use case after these changes.

Reply:
I have added these in the latest commit. I imagine there is a lot of ways we can make the example more robust and similar to common real use cases.

Reply:
If you create a new list in `Setup`, it does not make any sense to clear it in `Teardown`.

Reply:
> I imagine there is a lot of ways we can make the example more robust and similar to common real use cases.

Any options to make the example more realistic are welcome.

Reply:
Got it. I updated the @Teardown method to print to the console for now and will brainstorm ways to potentially revamp the example to reflect a more coherent use case that utilizes the libraries annotations effectively.